liblib/CMakeLists.txt

```diff
@@ -4,194 +4,226 @@
 # University Corporation for Atmospheric Research/Unidata.
 
 # See netcdf-c/COPYRIGHT file for more info.
-SET(liblib_LIBS dispatch netcdf3)
 
 #####
 # Add target objects/modules based on options.
 #####
-
-IF(USE_NETCDF4)
-  SET(liblib_LIBS ${liblib_LIBS} netcdf4)
-ENDIF()
-IF(USE_HDF5)
-  SET(liblib_LIBS ${liblib_LIBS} netcdfhdf5)
-ENDIF()
-
-IF(USE_PNETCDF)
-  SET(liblib_LIBS ${liblib_LIBS} netcdfp)
-ENDIF()
-
-IF(USE_HDF4)
-  SET(liblib_LIBS ${liblib_LIBS} netcdfhdf4)
-ENDIF()
-
-IF(ENABLE_DAP2)
-  SET(liblib_LIBS ${liblib_LIBS} oc2 dap2)
-ENDIF()
-
-IF(ENABLE_DAP4)
-  SET(liblib_LIBS ${liblib_LIBS} dap4)
-  SET(liblib_LIBS ${liblib_LIBS} ncxml)
-ENDIF()
-
-IF(ENABLE_NCZARR)
-  SET(liblib_LIBS ${liblib_LIBS} nczarr)
-ENDIF()
-
-IF(ENABLE_S3_INTERNAL)
-  SET(liblib_LIBS ${liblib_LIBS} ncxml)
-ENDIF()
-
-IF(ENABLE_PLUGINS)
-  SET(liblib_LIBS ${liblib_LIBS} ncpoco)
-ENDIF()
-
-FOREACH(LIBS ${liblib_LIBS})
-  SET(LARGS ${LARGS} $<TARGET_OBJECTS:${LIBS}>)
-ENDFOREACH()
-
-ADD_LIBRARY(netcdf nc_initialize.c ${LARGS} )
-
-IF(STATUS_PARALLEL)
+target_sources(netcdf
+  PRIVATE
+    $<TARGET_OBJECTS:netcdf3>
+    $<TARGET_OBJECTS:dispatch>
+)
+
+if(USE_NETCDF4)
+  target_sources(netcdf
+    PRIVATE
+      $<TARGET_OBJECTS:netcdf4>
+  )
+endif()
+
+if(USE_HDF5)
+  target_sources(netcdf
+    PRIVATE
+      $<TARGET_OBJECTS:netcdfhdf5>
+  )
+endif()
+
+if(USE_PNETCDF)
+  target_sources(netcdf
+    PRIVATE
+      $<TARGET_OBJECTS:netcdfp>
+  )
+endif()
+
+if(USE_HDF4)
+  target_sources(netcdf
+    PRIVATE
+      $<TARGET_OBJECTS:netcdfhdf4>
+  )
+endif()
+
+if(ENABLE_DAP2)
+  target_sources(netcdf
+    PRIVATE
+      $<TARGET_OBJECTS:oc2>
+      $<TARGET_OBJECTS:dap2>
+  )
+endif()
+
+if(ENABLE_DAP4)
+  target_sources(netcdf
+    PRIVATE
+      $<TARGET_OBJECTS:dap4>
+      $<TARGET_OBJECTS:ncxml>
+  )
+endif()
+
+if(ENABLE_NCZARR)
+  target_sources(netcdf
+    PRIVATE
+      $<TARGET_OBJECTS:nczarr>
+  )
+endif()
+
+if(ENABLE_S3_INTERNAL)
+  target_sources(netcdf
+    PRIVATE
+      $<TARGET_OBJECTS:ncxml>
+  )
+endif()
+
+if(ENABLE_PLUGINS)
+  target_sources(netcdf
+    PRIVATE
+      $<TARGET_OBJECTS:ncpoco>
+  )
+endif()
+
+target_sources(netcdf
+  PRIVATE
+    ${CMAKE_CURRENT_SOURCE_DIR}/nc_initialize.c 
+)
+
+if(STATUS_PARALLEL)
   # TODO: Make PUBLIC when other dependencies have PUBLIC/PRIVATE specified
   target_link_libraries(netcdf MPI::MPI_C)
-ENDIF(STATUS_PARALLEL)
+endif(STATUS_PARALLEL)
 
-IF(TARGET MPI::MPI_C)
+if(TARGET MPI::MPI_C)
   target_link_libraries(netcdf MPI::MPI_C)
```

Comment:
Why the repeat?

Reply:
Looking at `git blame`, I think this was from two separate PRs, so one of these can safely go